src/udev/udevadm-control.c

```diff
@@ -136,11 +137,12 @@ static int adm_control(struct udev *udev, int argc, char *argv[]) {
                 case 't': {
                         int seconds;
 
-                        seconds = atoi(optarg);
-                        if (seconds >= 0)
-                                timeout = seconds;
-                        else
+                        rc = safe_atoi(optarg, &seconds);
```

Comment:
Hmm, I think if we rework this we should change this to use parse_sec() instead, which is a superset of atoi() in this context, and understands the usual "h", "min", ... suffixes. (of course, it actually returns usec_t, instead of int, but converting that should be simple enough...